code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1450,24 +1450,28 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 
 	var/loc_temp = H.get_temperature(environment)
 
-	//Body temperature is adjusted in two steps. First, your body tries to stabilize itself a bit.
-	if(H.stat != DEAD)
-		H.natural_bodytemperature_stabilization()
-
-	//Then, it reacts to the surrounding atmosphere based on your thermal protection
+	//Body temperature is adjusted in two parts: first there your body tries to naturally preserve homeostasis (shivering/sweating), then it reacts to the surrounding environment
+	//Thermal protection (insulation) has mixed benefits in two situations (hot in hot places, cold in hot places) 
 	if(!H.on_fire) //If you're on fire, you do not heat up or cool down based on surrounding gases
-		if(loc_temp < H.bodytemperature)
-			//Place is colder than we are
-			var/thermal_protection = H.get_cold_protection(loc_temp) //This returns a 0 - 1 value, which corresponds to the percentage of protection based on what you're wearing and what you're exposed to.
-			if(thermal_protection < 1)
-				H.bodytemperature += min((1-thermal_protection) * ((loc_temp - H.bodytemperature) / BODYTEMP_COLD_DIVISOR), BODYTEMP_COOLING_MAX)
-		else
-			//Place is hotter than we are
-			var/thermal_protection = H.get_heat_protection(loc_temp) //This returns a 0 - 1 value, which corresponds to the percentage of protection based on what you're wearing and what you're exposed to.
-			if(thermal_protection < 1)
-				H.bodytemperature += min((1-thermal_protection) * ((loc_temp - H.bodytemperature) / BODYTEMP_HEAT_DIVISOR), BODYTEMP_HEATING_MAX)
-
-	// +/- 50 degrees from 310.15K is the 'safe' zone, where no damage is dealt.
+		var/natural = 0
+		if((abs(BODYTEMP_DEFAULT - H.bodytemperature) <= 25) && (abs(BODYTEMP_DEFAULT - loc_temp) <= 25))
+			return //Performance saver
+		if(H.stat != DEAD)
+			natural = H.natural_bodytemperature_stabilization()
+		if(loc_temp < H.bodytemperature) //Place is colder than we are
+			var/thermal_protection = H.get_cold_protection(loc_temp) + 1 //This returns a 1 - 2 value, which corresponds to the percentage of protection based on what you're wearing and what you're exposed to.
+			if(H.bodytemperature < BODYTEMP_DEFAULT) //we're cold, insulation helps us generate heat and will reduce the heat we lose to the environment
+				H.bodytemperature += (thermal_protection)*natural + max((2-thermal_protection) * ((loc_temp - H.bodytemperature) / BODYTEMP_COLD_DIVISOR), BODYTEMP_COOLING_MAX)
+			else //we're sweating, insulation hinders our ability to reduce heat - and it will reduce the amount of cooling you get from the environment
+				H.bodytemperature += natural*(1/thermal_protection) + max((3-thermal_protection) * ((loc_temp - H.bodytemperature) / BODYTEMP_COLD_DIVISOR), BODYTEMP_COOLING_MAX)
+		else //Place is hotter than we are
+			var/thermal_protection = H.get_heat_protection(loc_temp) + 1 //This returns a 1 - 2 value, which corresponds to the percentage of protection based on what you're wearing and what you're exposed to.
```

Comment:
elevate `thermal_protection = 1` to above L1461 and have it set within ifs (or within the proc) rather than having 2 separated vars.